addons/mrp/models/stock_move.py

```diff
@@ -487,6 +487,12 @@ def _key_assign_picking(self):
         keys = super(StockMove, self)._key_assign_picking()
         return keys + (self.created_production_id,)
 
+    def _search_picking_for_assignation_domain(self):
+        res = super()._search_picking_for_assignation_domain()
+        if self.created_production_id:
+            res = expression.AND([res[1:], ['|', res[0], ('move_ids.created_production_id', '=', self.created_production_id.id)]])
```

Comment:
This method is overrided in `stock_picking_batch`, adding a bunch of `'&'` in the domain and making this code very prompt to traceback

Reply:
After some tests, it seems it happens only when `mrp` is installed after `stock_picking_batch` and the server isn't rebooted meanwhile, so the call order is: `mrp` -> call super in `stock_picking_batch` -> call super in `stock` and then, when we are at this line, `res[0]` is `'&'`, so the second parameter for `expression.AND` is `['|', '&', ('move_ids.created_production_id', '=', self.created_production_id.id)]` which causes the traceback